Main.cs

```diff
@@ -66,6 +68,37 @@ private static bool OnToggle(UnityModManager.ModEntry modEntry, bool value)
             return true;
         }
 
+        private static void PatchPersistentJobs()
+        {
+            Type? persistentJobsModInteractionFeaturesType = GetPersistentJobsInteractionType();
+            if (persistentJobsModInteractionFeaturesType != null)
+            {
+                var jobTracksChanged = persistentJobsModInteractionFeaturesType.GetEvent("JobTracksChanged");
+                jobTracksChanged.AddEventHandler(null, new Action<Job>(JobData.JobPatches.UpdateJobsFromPersistentJobs));
+                DebugLog(() => "Persistent Jobs found and hooked");
+            }
+        }
+        /// <summary>
+        /// This will check to see if persistent jobs mod is currently installed and loaded
+        /// </summary>
+        /// <returns>
+        /// Persistent Jobs mod event interface if available, null otherwise.
+        /// </returns>
+        private static Type? GetPersistentJobsInteractionType()
+        {
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
```

Comment:
Please use `UnityModManager.FindMod()` to see if PersistentJobs is loaded before doing the reflection. You can then use the `ModEntry.Assembly` property to search just that assembly instead of every loaded assembly.

Reply:
Done